src/pipeline/activator/activator/planet/main.py

```diff
@@ -0,0 +1,275 @@
+        raise Exception('Imagery inactive')
+    else:
+        download_link = result['location']
+
+    return download_link
+
+
+def download_from_planet(download_link,
+                         planet_id,
+                         planet_api_key,
+                         temp_dir,
+                         download):
+    filename_tiff = f'/{temp_dir}/{planet_id}.tiff'
+    logger.info(f'Downloading imagery to {filename_tiff}')
+    if download:
+        data = requests.get(download_link,
+                            auth=HTTPBasicAuth(planet_api_key, ''),
+                            stream=True)
+        with open(filename_tiff, 'wb') as f:
+            data.raw.decode_content = True
+            shutil.copyfileobj(data.raw, f)
+        os.system(f'gdaladdo {filename_tiff}')
+    return filename_tiff
+
+
+def upload_to_s3(filename_tiff, s3_bucket, s3_prefix, upload):
+    filename = filename_tiff.split('/')[-1]
+    s3_uri = f's3://{s3_bucket}/{s3_prefix}/{filename}'
+    logger.info(f'Uploading imagery to {s3_uri}')
+    if upload:
+        os.system(f'aws s3 cp {filename_tiff} {s3_uri}')
+    return s3_uri
+
+
+def generate_stac_item(filename_tiff, cog_collection, planet_id, s3_uri):
+    logger.info(f'Using gdalinfo to get metadata')
+    filename_json = filename_tiff.replace('.tiff', '.json')
+    os.system(f'gdalinfo -proj4 -json {filename_tiff} > {filename_json}')
+    with open(filename_json, 'r') as f:
+        data = json.load(f)
+
+    logger.info(f'Organizing metadata')
+    tifftag_datetime = data.get('metadata').get('').get('TIFFTAG_DATETIME')
+    year, month, day = [
+        int(n) for n in tifftag_datetime.split(' ')[0].split(':')
+    ]
+    dt = datetime(year, month, day, tzinfo=timezone.utc)
+    polygon = data.get('wgs84Extent')
+    coords = polygon.get('coordinates')
+    while len(coords) == 1:
+        coords = coords[0]
+    ys = [y for (y, x) in coords]
+    xs = [x for (y, x) in coords]
+    bbox = [min(ys), min(xs), max(ys), max(xs)]
+    props = {
+        'eo:bands': cog_collection.properties['eo:bands'],
+        'hsi:wavelength_min': cog_collection.properties['hsi:wavelength_min'],
+        'hsi:wavelength_min': cog_collection.properties['hsi:wavelength_min'],
+    }
+
+    logger.info(f'Creating new cog item')
+    cog_item = pystac.Item(planet_id,
+                           polygon,
+                           bbox,
+                           dt,
+                           props,
+                           stac_extensions=COG_ITEM_EXTENSIONS,
+                           collection=cog_collection.id)
+    cog_item.add_asset(
+        'tiff_0',
+        pystac.Asset(s3_uri, media_type=pystac.MediaType.COG, roles=['data']))
+
+    return cog_item
+
+
+def update_franklin(cog_item, cog_collection, stac_api_uri, update):
+    logger.info('Updating Franklin')
+    stac_client = STACClient(stac_api_uri)
+    if False and update:  # XXX
+        if not stac_client.has_collection(cog_collection.id):
```

Comment:
There are two solutions to that: 1. path our client  - add methods you need (probably faster) 2. use the py stac stac api client (better and maintained as an OSS)